### ISSUE TITLE
Application not visible when deploying to other namespace than argocd

### ISSUE THREAD

**Issue by peterbosalliandercom** [argoproj/argo-cd#3847]
With argocd 1.6.1 If we create an application with a project other than default it does not become visible in the UI. 
1. create project x, with all permissions
2. create argocd-rbac-cm with all permissions (admin)
2. Deploy application to project x in other namespace (Y) than argocd is installed
3. try to find it on the UI. The app is in the namespace (Y) but not visible.

Should argocd list all applications in all namespaces by default or only the ones in argocd namespace?

In the operator manual it says this: 
  # You'll usually want to add your resources to the argocd namespace.
  namespace: argocd
https://argoproj.github.io/argo-cd/operator-manual/application.yaml

But is this compulsory? We have teams that use their own applications and do not have permissions to create applications in the argocd namespace. 

Checklist:

* [x] I've searched in the docs and FAQ for my answer: http://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Comment by peterbosalliandercom** [comment]
I have found this: https://github.com/argoproj/argo-cd/issues/2557 where Alexec says: In short, I'm afraid you cannot. Argo CD only listens to apps created in it's own namespace: argocd. This is entirely intentional. Consider the costs of listening to many namespaces for changes, or how you'd reliability do HA/DR.

**Comment by peterbosalliandercom** [comment]
We really need the multi namespace setup because we want to use argocd with rbac in a multitenant matter. We want a tenant to deploy their own applications in their own namespaces from argocd in a separate project. is this a new feature?

**Comment by nastacio** [comment]
Doesn't that mean all AppProject instances must have this destination in order to support app-of-apps?

```
  destinations:
    - name: in-cluster
      namespace: argocd-instance-namespace
      server: https://kubernetes.default.svc
```

And wouldn't that effectively allow an Application to manage anything in `clusterResourceWhitelist` inside that namespace?

**Comment by jasonhao518** [comment]
Looking forward the same feature, 

> We really need the multi namespace setup because we want to use argocd with rbac in a multitenant matter. We want a tenant to deploy their own applications in their own namespaces from argocd in a separate project. is this a new feature?

I'm looking forward the same feature, actually the idea is quite similar, when project and applications in a dedicate namespace of a tenant, means disposal will be quite simple, just need to drop everything in the namespace.  @peterbosalliandercom do you find any solution or work around for this ?

**Comment by crenshaw-dev** [comment]
@jasonhao518 there is a PR for this feature. Please review and comment if you have time! https://github.com/argoproj/argo-cd/pull/9755

**Comment by JasonHao123** [comment]
> @jasonhao518 there is a PR for this feature. Please review and comment if you have time! #9755

Thanks a lot for your invitation. Already commented on the ticket

**Comment by jaredmcqueen** [comment]
fyi this would be a great feature and would help assist with any related tasks to get it merged.  Does anything else need to be done?

**Comment by crenshaw-dev** [comment]
@jaredmcqueen it's merged and will be part of 2.5! Any help testing either the `latest` build or 2.5.0-rc1 when it comes out would be very appreciated.

I believe #9755 was also very light on docs. So if you test and happen to write up some docs, I'd be more than happy to review a PR.  :-)

**Comment by nxhuy-github** [comment]
@crenshaw-dev When would the release 2.5 be available, please?
And I feel like the fact that the `sourceNamespaces` is not available too in argocd v2.4.11, no? Because I tried to add `sourceNamespaces` in my `AppProject` but it always says: `<nil>: Invalid value: "The edited file failed validation": ValidationError(AppProject.spec): unknown field "sourceNamespaces" in io.argoproj.v1alpha1.AppProject.spec.`
So in the waiting time, how can we solve the problem, please?
Thank you

**Comment by vl-kp** [comment]
> @crenshaw-dev When would the release 2.5 be available, please? And I feel like the fact that the `sourceNamespaces` is not available too in argocd v2.4.11, no? Because I tried to add `sourceNamespaces` in my `AppProject` but it always says: `<nil>: Invalid value: "The edited file failed validation": ValidationError(AppProject.spec): unknown field "sourceNamespaces" in io.argoproj.v1alpha1.AppProject.spec.` So in the waiting time, how can we solve the problem, please? Thank you

got the same error, how to solve it

**Comment by NikolaiBessonov** [comment]
> > @crenshaw-dev When would the release 2.5 be available, please? And I feel like the fact that the `sourceNamespaces` is not available too in argocd v2.4.11, no? Because I tried to add `sourceNamespaces` in my `AppProject` but it always says: `<nil>: Invalid value: "The edited file failed validation": ValidationError(AppProject.spec): unknown field "sourceNamespaces" in io.argoproj.v1alpha1.AppProject.spec.` So in the waiting time, how can we solve the problem, please? Thank you
> 
> got the same error, how to solve it

have you solved the problem?
